src/kernels/execution/lastCellExecutionTracker.ts

```diff
@@ -33,18 +34,21 @@ export class LastCellExecutionTracker extends Disposables implements IExtensionS
     public activate(): void {
         this.serverStorage.onDidRemove(this.onDidRemoveServerUris, this, this.disposables);
     }
-    private getStateKey(serverId: string) {
-        return `LAST_EXECUTED_CELL_${serverId}`;
+    private async getStateKey(serverHandle: JupyterServerProviderHandle) {
+        return `LAST_EXECUTED_CELL_${await computeServerId(serverHandle)}`;
```

Comment:
We have left this for backwards compatibility, computeServerid is not used in most places, 
but this is an exception as we have stored state in older versions of extension and we should preserve that.